flexmeasures/api/v3_0/tests/test_sensors_api.py

```diff
@@ -43,54 +49,46 @@ def test_fetch_sensors(
     Our user here is admin, so is allowed to see all sensors.
     Pagination is tested only in passing, we should test filtering and page > 1
     """
-    query = {}
+    query = {search_by: search_value}
+
     if use_pagination:
         query["page"] = 1
 
-    if search_by:
-        query["unit"] = search_by
+    if search_by == "unit":
+        query["unit"] = search_value
+    elif search_by == "filter":
+        query["filter"] = search_value.split(" ")
 
     if all_accessible:
         query["all_accessible"] = True
 
-    if search_value:
-        # get sensors from test data
-        dict_key = list(setup_api_test_data.keys())[1]
-        sensor = setup_api_test_data[dict_key]
-        keywords = sensor.name.split(" ")
-        query["filter"] = keywords  # filter expects an array of strings
-
     response = client.get(
         url_for("SensorAPI:index"),
         query_string=query,
     )
 
+    print("Query:\n%s" % query)
     print("Server responded with:\n%s" % response.json)
     assert response.status_code == 200
 
-    if search_by:
-        assert is_valid_unit(response.json[0]["unit"])
-        assert response.json[0]["unit"] == "°C"
-
     if use_pagination:
         assert isinstance(response.json["data"][0], dict)
         assert is_valid_unit(response.json["data"][0]["unit"])
-        print(
-            "Num records: %s" % response.json["num-records"],
-            response.json["filtered-records"],
-        )
-        if search_value:
-            # NOTE: Even if the entire sensor name was passed to filter, due to the nature of the filter
-            # logic, the response will contain all sensors that have most of the keywords in the name
-            assert response.json["data"][1]["name"] == sensor.name
-            assert response.json["num-records"] == num_sensors
-            assert response.json["filtered-records"] == num_sensors
-        else:
-            assert response.json["num-records"] == num_sensors
-            assert response.json["filtered-records"] == num_sensors
+        assert response.json["num-records"] == 3
+        assert response.json["filtered-records"] == 3
     else:
         assert isinstance(response.json, list)
         assert is_valid_unit(response.json[0]["unit"])
+        # NOTE: Even if the entire sensor name was passed to filter, due to the nature of the filter logic, the response will contain
+        # all sensors that have most of the keywords in the name. That's why we have 2 sensors in the response and I'm selecting the
```

Comment:
I need to understand this better. With "most of the keywords" you mean all of the keywords right? So we'd get both

- "some temperature sensor" and
- "some other temperature sensor"
because all three keywords are in there.

I believe we might be able to use quotes a string like `'some temperature sensor'` to search and then this might be different.

If I might be right, please test. That would make this test easier to read.

Reply:
No not necessarily all, for example 2 out of 3 o the keywords

Reply:
i will test your suggestion

Reply:
> I believe we might be able to use quotes a string like 'some temperature sensor' to search and then this might be different.

Yes. The `SearchFilterField` should handle quoted search terms as one term possibly containing spaces.

Reply:
This works thanks for the suggestion @nhoening